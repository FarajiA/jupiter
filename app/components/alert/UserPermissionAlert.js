import React from 'react';
import Alert from '../helix/status/Alert';
import { useTranslation } from 'react-i18next';

const UserPermissionAlert = (props) => {
  const { t } = useTranslation();
  const submit = () => {
    window.open('https://one.rackspace.com/display/IDPLAT/Jupiter+-+Internal+Cart', '_blank');
  };
  return (
    <Alert
      className="u-space-top"
      type="error"
      static
      show
      persist="true"
      cta={t('account:user.tip.viewDocumentation')}
      onSubmit={submit}
    >
      <p>{t('account:user.tip.invalidAccountPermission')}</p>
      <small>
        {t('account:actions.jupiter.goToProductGuide')}
      </small>
    </Alert>
  );
};

export default UserPermissionAlert;
