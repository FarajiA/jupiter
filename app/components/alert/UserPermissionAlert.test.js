import React from 'react';
import enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import UserPermissionAlert from './UserPermissionAlert';

describe('UserPermissionAlert', () => {
  global.window.open = jest.fn();
  const shallow = (props) => enzyme.shallow(<UserPermissionAlert {...props} />);
  test('UserPermissionAlert', () => {
    const component = renderer.create(<UserPermissionAlert />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('it invokes window.open()', () => {
    const wrapper = shallow();
    wrapper.find('Alert').props().onSubmit();
    expect(global.window.open).toHaveBeenCalled();
  });
});
