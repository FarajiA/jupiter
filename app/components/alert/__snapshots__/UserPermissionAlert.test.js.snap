// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPermissionAlert UserPermissionAlert 1`] = `
<div>
  <hx-alert
    class="u-space-top"
    cta="View Docs"
    onDismiss={[Function]}
    onSubmit={[Function]}
    persist="true"
    show={true}
    static={true}
    type="error"
  >
     
    <p>
      You do not have the required permissions to access this resource.
    </p>
    <small>
      For information on how to request access, refer to the documentation.
    </small>
  </hx-alert>
</div>
`;
